File: fixture-app-go/discount_test.go
package main

import (
	"net/http"
	"net/http/httptest"
	"testing"
)

// TestHandleBulkDiscountAppliesAtExactlyTenUnits ensures the 10% bulk
// discount is applied for orders of exactly 10 units, not just orders
// strictly greater than 10.
func TestHandleBulkDiscountAppliesAtExactlyTenUnits(t *testing.T) {
	req := httptest.NewRequest(http.MethodGet, "/discount?unit_price=100&qty=10", nil)
	w := httptest.NewRecorder()

	handleBulkDiscount(w, req)

	got := w.Body.String()
	want := "900"
	if got != want {
		t.Errorf("handleBulkDiscount(unit_price=100, qty=10) = %q, want %q", got, want)
	}
}

func TestHandleBulkDiscountNoDiscountUnderTen(t *testing.T) {
	req := httptest.NewRequest(http.MethodGet, "/discount?unit_price=50&qty=9", nil)
	w := httptest.NewRecorder()

	handleBulkDiscount(w, req)

	got := w.Body.String()
	want := "450"
	if got != want {
		t.Errorf("handleBulkDiscount(unit_price=50, qty=9) = %q, want %q", got, want)
	}
}

File: fixture-app-node/discount.test.js
const { bulkDiscount } = require("./discount");

describe("bulkDiscount", () => {
  test("applies 10% discount to orders of exactly 10 units", () => {
    // unitPrice=100, qty=10 -> total=1000, discounted=900
    expect(bulkDiscount(100, 10)).toBe(900);
  });

  test("applies 10% discount to orders of more than 10 units", () => {
    // unitPrice=100, qty=12 -> total=1200, discounted=1080
    expect(bulkDiscount(100, 12)).toBe(1080);
  });

  test("does not discount orders under 10 units", () => {
    // unitPrice=100, qty=9 -> total=900, no discount applied
    expect(bulkDiscount(100, 9)).toBe(900);
  });
});
